docs/dev-guide.md

```diff
@@ -4,13 +4,15 @@ In general we prefer simplicity. We standardize on JavaScript/TypeScript (Node.j
 
 ## High level view
 
-![High level system structure](high-level.png)
+```d2
+--8<-- "docs/diagrams/high-level.d2"
```

Comment:
Shrunk it down, but my screen is certainly much smaller. Probably need some custom CSS for a `max-width` on diagrams.

Reply:
This is what it's looking like for me at the moment. A `max-width` on this would be great!

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/3710a51b-dd5c-4c11-b8e5-53fc6ad71c95" />

Reply:
Did a max-height limit, I think that is more useful than a max-width.

Reply:
Somewhat better, but still huge, and still with the extra whitespace at the top:

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/c5130403-2b64-479c-bf23-1cf1f26c14f7" />

Reply:
Fixed all these up to the extent I can.